test/membrane_live_web/plug/auth_plug_integration_test.exs

```diff
@@ -0,0 +1,112 @@
+defmodule MembraneLiveWeb.AuthPlugIntegrationTest do
+  use MembraneLiveWeb.ConnCase
+
+  import MembraneLive.WebinarsFixtures
+  import MembraneLive.AccountsFixtures
+
+  alias MembraneLive.Support.AuthTokenMock
+
+  @dummy_uuid "5a2771ef-3cf2-4d86-b125-fd366e04bc29"
+
+  @endpoint_atoms [:create, :index, :update, :delete, :show]
+  @arity_2_endpoints [:create, :index]
+  @arity_3_endpoints [:update, :delete, :show]
+
+  setup %{conn: conn} do
+    conn_with_wrong_bearer =
+      conn
+      |> put_req_header("accept", "application/json")
+      |> put_req_header("authorization", get_invalid_bearer())
+
+    {:ok, conn: conn_with_wrong_bearer}
+  end
+
+  describe "webinar" do
+    @webinar_endpoint_infos [
+      %{endpoint_atom: :create, body: %{webinar: webinar_attrs()}},
+      %{endpoint_atom: :update, body: %{webinar: webinar_attrs()}},
+      %{endpoint_atom: :delete},
+      %{endpoint_atom: :show},
+      %{endpoint_atom: :index}
+    ]
+
+    Enum.map(@webinar_endpoint_infos, fn %{endpoint_atom: atom} = info ->
+      test "[401] #{atom} webinar", %{conn: conn_with_wrong_bearer} do
+        test_webinar_endpoint(conn_with_wrong_bearer, unquote(Macro.escape(info)))
+      end
+    end)
+
+    defp test_webinar_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_2_endpoints do
+      test_endpoint(&Routes.webinar_path/2, conn, endpoint_info)
+    end
+
+    defp test_webinar_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_3_endpoints do
+      test_endpoint(&Routes.webinar_path/3, conn, endpoint_info)
+    end
+  end
+
+  describe "user" do
+    @user_endpoint_infos [
+      %{endpoint_atom: :update, body: %{user: user_attrs()}},
+      %{endpoint_atom: :delete},
+      %{endpoint_atom: :show},
+      %{endpoint_atom: :index}
+    ]
+
+    Enum.map(@user_endpoint_infos, fn %{endpoint_atom: atom} = info ->
+      test "[401] user #{atom} resources test", %{conn: conn_with_wrong_bearer} do
+        test_user_endpoint(conn_with_wrong_bearer, unquote(Macro.escape(info)))
+      end
+    end)
+
+    defp test_user_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_2_endpoints do
+      test_endpoint(&Routes.user_path/2, conn, endpoint_info)
+    end
+
+    defp test_user_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_3_endpoints do
+      test_endpoint(&Routes.user_path/3, conn, endpoint_info)
+    end
+  end
+
+  defp test_endpoint(routing_function, conn, %{endpoint_atom: atom} = endpoint_info)
+       when atom in @endpoint_atoms do
```

Comment:
This `when` ensures that we won't get further into the function. Although in this case this wouldn't change a lot (since  `endpoint_atom_to_function` whould raise `no_function_clause` as well), it's nice to define, which atoms do we want and which are implemented.